### PR TITLE
Refactor security code

### PR DESCRIPTION
**Pull request by ggezer**

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java [hunk 1]
```diff
@@ -74,7 +75,7 @@ public void checkParentPermission(Mode.Bits bits, LockedInodePath inodePath)
 
     // collects existing inodes info on the path. Note that, not all the components of the path have
     // corresponding inodes.
-    List<Inode> inodeList = inodePath.getInodeList();
+    List<InodeView> inodeList = (List<InodeView>) (List<?>) inodePath.getInodeList();
```
Comment (bf8086) [inline core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java]: This looks like a revert of #8307. Is it intentional?

File: core/server/worker/src/main/java/alluxio/worker/grpc/AbstractReadHandler.java [hunk 2]
```diff
@@ -90,16 +93,19 @@
...
+      StreamObserver<ReadResponse> responseObserver, AuthenticatedUserInfo userInfo) {
     mDataReaderExecutor = executorService;
     mSerializingExecutor = new SerializingExecutor(executorService);
     mResponseObserver = responseObserver;
+    mUserInfo = userInfo;
   }
 
   @Override
   public void onNext(alluxio.grpc.ReadRequest request) {
+
```
Comment (bf8086) [inline core/server/worker/src/main/java/alluxio/worker/grpc/AbstractReadHandler.java]: (nit) redundant new line?

File: core/server/worker/src/main/java/alluxio/worker/grpc/BlockWorkerImpl.java [hunk 3]
```diff
@@ -29,13 +29,17 @@
...
+import alluxio.security.authentication.AuthenticatedUserInfo;
 import alluxio.util.IdUtils;
+import alluxio.util.SecurityUtils;
 import alluxio.worker.WorkerProcess;
 import alluxio.worker.block.AsyncCacheRequestManager;
 import alluxio.worker.block.BlockWorker;
 
 import com.google.common.collect.ImmutableMap;
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.grpc.Status;
```
Comment (bf8086) [inline core/server/worker/src/main/java/alluxio/worker/grpc/BlockWorkerImpl.java]: Sort import?

File: core/server/worker/src/main/java/alluxio/worker/grpc/BlockWriteHandler.java [hunk 4]
```diff
@@ -18,6 +18,7 @@
 import alluxio.grpc.WriteResponse;
 import alluxio.metrics.MetricsSystem;
 import alluxio.metrics.WorkerMetrics;
+import alluxio.security.authentication.AuthenticatedUserInfo;
```
Comment (bf8086) [inline core/server/worker/src/main/java/alluxio/worker/grpc/BlockWriteHandler.java]: Sort import?

File: core/server/worker/src/main/java/alluxio/worker/grpc/DelegationWriteHandler.java [hunk 5]
```diff
@@ -29,15 +30,18 @@
   private final WorkerProcess mWorkerProcess;
   private final DataMessageMarshaller<WriteRequest> mMarshaller;
   private AbstractWriteHandler mWriteHandler;
+  private alluxio.security.authentication.AuthenticatedUserInfo mUserInfo;
```
Comment (bf8086) [inline core/server/worker/src/main/java/alluxio/worker/grpc/DelegationWriteHandler.java]: ```private AuthenticatedUserInfo mUserInfo;```

File: core/server/worker/src/main/java/alluxio/worker/grpc/ShortCircuitBlockReadHandler.java [hunk 6]
```diff
@@ -48,17 +49,20 @@
   /** The lock Id of the block being read. */
   private long mLockId;
   private long mSessionId;
+  private alluxio.security.authentication.AuthenticatedUserInfo mUserInfo;
```
Comment (bf8086) [inline core/server/worker/src/main/java/alluxio/worker/grpc/ShortCircuitBlockReadHandler.java]: ```  private AuthenticatedUserInfo mUserInfo;```

File: core/server/worker/src/main/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandler.java [hunk 7]
```diff
@@ -52,15 +53,19 @@
 
   private long mSessionId = INVALID_SESSION_ID;
 
+  private alluxio.security.authentication.AuthenticatedUserInfo mUserInfo;
```
Comment (bf8086) [inline core/server/worker/src/main/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandler.java]: ```private AuthenticatedUserInfo mUserInfo;```

File: core/server/worker/src/test/java/alluxio/worker/grpc/AbstractWriteHandlerTest.java [hunk 8]
```diff
@@ -24,6 +24,7 @@
 import alluxio.grpc.WriteRequestCommand;
 import alluxio.grpc.WriteResponse;
 import alluxio.network.protocol.databuffer.DataBuffer;
+import alluxio.security.authentication.AuthenticatedUserInfo;
```
Comment (bf8086) [inline core/server/worker/src/test/java/alluxio/worker/grpc/AbstractWriteHandlerTest.java]: Sort import?

File: job/server/src/main/java/alluxio/master/AlluxioJobMasterProcess.java [hunk 9]
```diff
@@ -79,6 +79,15 @@
     }
   }
 
+  @Override
+  public <T extends Master> T getMaster(Class<T> clazz) {
+    if (clazz == JobMaster.class) {
+      return (T) mJobMaster;
+    } else {
+      throw new RuntimeException(String.format("Could not found the master: %s", clazz));
```
Comment (bf8086) [inline job/server/src/main/java/alluxio/master/AlluxioJobMasterProcess.java]: ```Could not found``` => ```Could not find```

File: core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java [hunk 10]
```diff
@@ -108,6 +108,7 @@ public static BlockInStream create(FileSystemContext context, BlockInfo info,
         ReadRequest.newBuilder().setBlockId(blockId).setPromote(readType.isPromote());
     // Add UFS fallback options
     builder.setOpenUfsBlockOptions(options.getOpenUfsBlockOptions(blockId));
+
```
[the remark this answers is not in the document]
Comment (ggezer) [reply core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java]: For consistency.

File: core/client/fs/src/main/java/alluxio/client/block/stream/LocalFileDataReader.java [hunk 11]
```diff
@@ -101,6 +101,7 @@ public void close() throws IOException {
     private final long mLocalReaderChunkSize;
     private final int mReadBufferSize;
     private final GrpcBlockingStream<OpenLocalBlockRequest, OpenLocalBlockResponse> mStream;
+
```
[the remark this answers is not in the document]
Comment (ggezer) [reply core/client/fs/src/main/java/alluxio/client/block/stream/LocalFileDataReader.java]: For consistency.

**Comment by ggezer** [comment]
@bf8086 Addressed your comments. PTAL.

File: core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java (hunk 1 above)
Comment (bf8086) [reply core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java]: ping?

Comment (ggezer) [reply core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java]: Honestly can't remember why did I do that. Reverted.

File: core/server/worker/src/main/java/alluxio/worker/grpc/AbstractWriteHandler.java [hunk 12]
```diff
@@ -19,6 +19,7 @@
 import alluxio.grpc.WriteRequest;
 import alluxio.grpc.WriteRequestCommand;
 import alluxio.grpc.WriteResponse;
+import alluxio.security.authentication.AuthenticatedUserInfo;
```
Comment (bf8086) [inline core/server/worker/src/main/java/alluxio/worker/grpc/AbstractWriteHandler.java]: (nit) sort import

**Comment by ggezer** [comment]
@bf808 Updated with fixes and additional changes. PTAL.

**Comment by gpang** [comment]
@ggezer I think this looks good. My main question is what is the purpose for `GrpcServerAddress`?

**Comment by bf8086** [comment]
@gpang `GrpcServerAddress` is for exposing worker hostname to channel authenticator.

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Pull request merged**